daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_functions/Functions.tdml

```diff
@@ -4324,6 +4324,8 @@
       </xs:complexType>
     </xs:element>
     
+    <xs:element name="ceil08" type="xs:decimal" dfdl:inputValueCalc="{ fn:ceiling(xs:decimal(0.125)) }"/>
```

Comment:
You can test my theory about the side-effect of setScale by creating a test with two elements. The first is a bigDecimal type with a fractional part. The second is dfdl:inputValueCalc=".... access first element and round it....". If setScale modifies the object, then BOTH elements will have had the scale set.

Reply:
I'll add the test, but I think that it should be OK since setScale actually returns a copy with the scale set to the new value.

Reply:
Grrrr. I guess I'm not surprised.  Numbers are supposed to be immutable, so I'm very glad this is the case.

(I think it was unwise to prefix the name with "set..." if it returns a new value. "rescale" would have been a perfectly good name. I guess whoever created the setScale() method got it through code-review without someone raising this objection. Oh well. )

Reply:
I agree about using "set...", but then again common sense would also lead people to believe you should use the "round" function for rounding, so I'm not sure they were really going for common sense.